apis/workloads/v1alpha1/lifecycle_recorder.go

```diff
@@ -208,3 +221,38 @@ func (l *lifecycleEventRecorderImpl) ConsoleTerminate(ctx context.Context, csl *
 	l.logger.Info("event recorded", "id", id, "event", events.EventTerminated)
 	return nil
 }
+
+func appendStatusMessages(result map[string]string, containers []corev1.ContainerStatus) {
+	if containers == nil {
+		return
+	}
+
+	for _, containerStatus := range containers {
+		if containerStatus.State.Terminated != nil {
+			s := containerStatus.State.Terminated
```

Comment:
What is the state of a container that is in crashback loopoff? Will it be caught by this? Or is it handled by the waiting state below?

Reply:
I've just checked: looks like it'll be caught by waiting below

Reply:
Yes here is a live example that was generated in lab:
https://gocardless.atlassian.net/browse/TCSR-290
The app container is in waiting state.

Reply:
Well, not exactly a crashbackoff loop. It's an imagepullbackoff loop.

Reply:
Nice! Thanks for checking 😊